lib/widgets/payment_request_dialog.dart

```diff
@@ -466,6 +542,7 @@ class PaymentRequestDialogState extends State<PaymentRequestDialog>
     if (toPay > 0 && account.maxAllowedToPay >= toPay) {
       actions.add(SimpleDialogOption(
         onPressed: (() async {
+          _afterLayout(null);
```

Comment:
I think it is better to calculate the dialog height on payment fulfilled rather than here.

Reply:
We are using it to set the height of the consequent dialogs

Reply:
> We are using it to set the height of the consequent dialogs

I understand that. I was trying to say that since the heights are only used after payment fulfilled, we can apply all these calculation at that point and make all the variables local to that context instead of calling here _afterLayout and maintain a class variable.

Reply:
I'm a bit lost. Heights are not only used after payment fulfilled.
First dialog takes as much space as it needs and it's always bigger than the following dialogs(confirmation, processing payment). When user presses approve, we call _afterLayout() to calculate the size of the initial dialog to set the height of following dialogs. So they are same in size.